File: app/src/test/java/com/kirchhoff/example/githubclient/presenters/SplashPresenterTest.java
package com.kirchhoff.example.githubclient.presenters;

import com.kirchhoff.example.githubclient.Constants;
import com.kirchhoff.example.githubclient.DataModule;
import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage;
import com.kirchhoff.example.githubclient.ui.splash.SplashContract;
import com.kirchhoff.example.githubclient.ui.splash.SplashPresenter;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

import static junit.framework.Assert.assertNotNull;

/**
 * @author Kirchhoff-
 */

@RunWith(JUnit4.class)
public class SplashPresenterTest {

    @Mock
    SplashContract.View view;

    KeyValueStorage storage;

    private SplashPresenter presenter;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);

        storage = DataModule.provideKeyValueStorage();

        presenter = new SplashPresenter(storage, view);
    }

    @Test
    public void testCreated() throws Exception {
        assertNotNull(presenter);
        Mockito.verifyNoMoreInteractions(view);
    }

    @Test
    public void testEmptyUserName() throws Exception {
        storage.saveUserName(Constants.EMPTY_USER_NAME);

        presenter.subscribe();
        Thread.sleep(2200);
        Mockito.verify(view).openAuthScreen();
        Mockito.verifyNoMoreInteractions(view);
    }

    @Test
    public void testExistUserName() throws Exception {
        storage.saveUserName(Constants.USER_NAME);

        presenter.subscribe();
        Thread.sleep(2200);
        Mockito.verify(view).openRepositoriesScreen();
        Mockito.verifyNoMoreInteractions(view);
    }
}

File: app/src/test/java/com/kirchhoff/example/githubclient/presenters/SplashPresenterTest.kt
package com.kirchhoff.example.githubclient.presenters

import com.kirchhoff.example.githubclient.Constants
import com.kirchhoff.example.githubclient.Injection
import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage
import com.kirchhoff.example.githubclient.ui.splash.SplashContract
import com.kirchhoff.example.githubclient.ui.splash.SplashPresenter
import org.junit.Assert.assertNotNull
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.Mockito.verifyNoMoreInteractions
import org.mockito.MockitoAnnotations

/**
 * @author Kirchhoff-
 */
class SplashPresenterTest {

    @Mock
    private lateinit var view: SplashContract.View

    private lateinit var presenter: SplashPresenter

    private lateinit var storage: KeyValueStorage

    @Before
    @Throws(Exception::class)
    fun init() {
        MockitoAnnotations.initMocks(this)

        storage = Injection.provideKeyValueStorage()

        presenter = SplashPresenter(storage, view)
    }

    @Test
    @Throws(Exception::class)
    fun testCreated() {
        assertNotNull(presenter)
        verifyNoMoreInteractions(view)
    }

    @Test
    @Throws(Exception::class)
    fun testEmptyUserName() {
        storage.saveUserName(Constants.EMPTY_USER_NAME)

        presenter.subscribe()
        Thread.sleep(2200)

        verify(view).openAuthScreen()
        verifyNoMoreInteractions(view)
    }

    @Test
    @Throws
    fun testExistUserName() {
        storage.saveUserName(Constants.USER_NAME)

        presenter.subscribe()
        Thread.sleep(2200)

        verify(view).openRepositoriesScreen()
        verifyNoMoreInteractions(view)
    }
}